firmware/powerboard2/code/app/CAN_componentSpecific.c

```diff
@@ -0,0 +1,45 @@
+/*
+ * CAN_componentSpecific.c
+ *
+ *  Created on: Aug 8, 2020
+ *      Author: Poornachander
+ */
+
+#include "CAN.h"
+#include "messageHandler.h"
+
+static void CAN_enablePeripheralsClockCallback(void);
+static void CAN_messageReceivedCallback(const protocol_MID_E messageID, const protocol_allMessages_U * const message);
+
+static const CAN_HWConfig_S CAN_HWConfig =
+{
+	// GPIO
+	.rxPin = 8,
+	.txPin = 9,
+	.GPIOPort = GPIOB,
+	.AFNumber = GPIO_AF8_CAN1,
+
+	// CAN
+	.CANPeriph = CAN1,
+
+	// Common
+	.enablePeripheralsClockCallback = CAN_enablePeripheralsClockCallback,
+};
+
+extern const CAN_config_S CAN_config;
+const CAN_config_S CAN_config =
+{
+	.HWConfig = &CAN_HWConfig,

```

Comment:
What is this dot before the variable names? I thought C does not allow that.

Reply:
It's how you access the member of a struct. `HWConfig` is a member of the struct `CAN_config_S`.